data/src/frame.rs

```diff
@@ -41,10 +43,52 @@ pub struct VideoInfo {
     pub pic_type: PictureType,
     pub width: usize,
     pub height: usize,
+    pub flipped: bool,
     pub format: Arc<Formaton>,
 }
 
+impl fmt::Display for VideoInfo {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}x{}", self.width, self.height)
+    }
+}
+
 impl VideoInfo {
+    /// Constructs a new `VideoInfo` instance.
+    pub fn new(w: usize, h: usize, flip: bool, pic_type: PictureType, fmt: Arc<Formaton>) -> Self {
+        let bits = fmt.get_total_depth();
+        VideoInfo {
+            pic_type,
+            width: w,
+            height: h,
+            flipped: flip,
+            format: fmt,
+        }
+    }
+    /// Returns picture width.
+    pub fn get_width(&self) -> usize {
+        self.width as usize
+    }
+    /// Returns picture height.
+    pub fn get_height(&self) -> usize {
+        self.height as usize
+    }
+    /// Returns picture orientation.
```

Comment:
There are multiple axes along which flipping could happen, so maybe make this `hfipped`, `vflipped` or similar to make it clearer?